datafusion/core/src/datasource/listing/table.rs

```diff
@@ -167,6 +187,14 @@ impl ListingTableConfig {
         })
     }
 
+    fn is_enable_url_table(session: &SessionState) -> bool {
+        session
+            .catalog_list()
+            .as_any()
+            .downcast_ref::<DynamicFileCatalog>()
```

Comment:
The only thing I am concerned about here is that this is a special case for DynamicFileCatalog which is tightly coupling the catalog to the ListingTable (when the hope is that we can eventually split them apart)

Would it be possible instead to add an explicit option to `ListingTableConfig` like `infer_partitions_from_path`

And then set that option when creating the `ListingTableConfig` in `DynamicTableProvider`?

That way this code will still work but we won't have a direct dependency on DynamicTableProvider